### ISSUE TITLE
Support for SnippetTextEdit

### ISSUE THREAD

**Issue by mickaelistria** [eclipse-lsp4j/lsp4j#810]
See https://github.com/microsoft/language-server-protocol/pull/1892/files
SnippetTextEdit is one of the most important features of LSP for a long time as it enables much more convenient workflows for refactorings.

**Comment by nixel2007** [comment]
If I understand it correctly, it is not even in master branch of specification yet.

**Comment by mickaelistria** [comment]
> If I understand it correctly, it is not even in master branch of specification yet.

Yes, it got merged in master yesterday with https://github.com/microsoft/language-server-protocol/pull/1892 and will be part of upcoming LSP 3.18 release.

**Comment by rgrunber** [comment]
If no one is working on this, maybe @hopehadfield might be interested in contributing back the custom classes in JDT-LS she has written/is writing to fill the lack of support in LSP4J.

**Comment by cdietrich** [comment]
Is there an eta for 3.18
